### PR TITLE
Add I3C support for FRDM-MCXN947

### PR DESCRIPTION
**Pull request by mmahadevan108**
Added driver for the NXP P3T1755 digital temperature sensor to test the I3C support. Tested with the thermometer sample

```
*** Booting Zephyr OS build v3.7.0-4343-gc018bab964fd ***                                                                       
Thermometer Example (arm)                                                                                                       
Temperature device is 0x1000a144, name is p3t1755@4800000236152a0090                                                            
Temperature is 29.1°C                                                                                                           
Temperature is 29.1°C                                                                                                           
```                                                                                                     
The sensor driver currently does not support monitoring the Threshold levels and generating an interrupt. This feature will be added later.

### CODE REVIEW

File: boards/nxp/frdm_mcxn947/board.c [hunk 1]
```diff
@@ -146,6 +146,9 @@ static int frdm_mcxn947_init(void)
 	CLOCK_SetClkDiv(kCLOCK_DivPLL1Clk0, 1U);
 #endif
 
+	/* Enable 1MHz clock. */
+	SYSCON->CLOCK_CTRL |= SYSCON_CLOCK_CTRL_FRO1MHZ_CLK_ENA_MASK;
```
Comment (danieldegrasse) [inline boards/nxp/frdm_mcxn947/board.c:150]: Why do we need this clock?

Comment (mmahadevan108) [reply boards/nxp/frdm_mcxn947/board.c:150]: Yes, this is needed. I could move this to the I3C specific code. However I suspect there are other peripherals that would also need this clock. Hence added it here.

Comment (danieldegrasse) [reply boards/nxp/frdm_mcxn947/board.c:150]: I think we should keep this conditional under the I3C specific code until multiple consumers exist

**Review by ubieda — DISMISSED** [review]
Nit-pick: adding a sensor driver just to test the I3C support on a board in a single PR seems a bit odd to me. However, driver looks good.

Approving assuming Daniel's comment will be addressed.

File: boards/nxp/frdm_mcxn947/board.c (hunk 1 above)
Comment (mmahadevan108) [reply boards/nxp/frdm_mcxn947/board.c]: Updated

**Review by ubieda — DISMISSED** [review]

**Review by danieldegrasse — DISMISSED** [review]

**Comment by mmahadevan108** [comment]
@MaureenHelm, can you help review this PR.

File: drivers/sensor/nxp/p3t1755/p3t1755.c [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+LOG_MODULE_REGISTER(P3T1755, CONFIG_SENSOR_LOG_LEVEL);
+
+static int p3t1755_read_reg(const struct device *dev, uint8_t reg, uint8_t *value, uint8_t len)
+{
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i3c)
+	struct p3t1755_data *data = dev->data;
+
+	return i3c_burst_read(data->i3c_dev, reg, value, len);
+#else
```
Comment (MaureenHelm) [inline drivers/sensor/nxp/p3t1755/p3t1755.c:23]: Will this work if there are two instances of the sensor in the system, one on an I3C bus and one on an I2C bus? The ifdef checks if *any* instance of the sensor is on an I3C bus, not if *this* instance of the sensor is on an I3C bus.

File: drivers/sensor/nxp/p3t1755/p3t1755.c [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i3c)
+	if (config->i3c.bus != NULL) {
+		data->i3c_dev = i3c_device_find(config->i3c.bus, &config->i3c.dev_id);
+		if (data->i3c_dev == NULL) {
+			LOG_ERR("Cannot find I3C device descriptor");
+			return -ENODEV;
+		}
+	}
+#else
+	if (!device_is_ready(config->bus_cfg.i2c.bus)) {
```
Comment (MaureenHelm) [inline drivers/sensor/nxp/p3t1755/p3t1755.c:120]: Use `i2c_is_ready_dt`

File: samples/sensor/thermometer/boards/frdm_mcxn947_mcxn947_cpu0.overlay [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+		ambient-temp0 = &p3t1755;
+	};
+};
+
+&i3c1 {
+	status = "okay";
+
+	i2c-scl-hz = <400000>;
+	i3c-scl-hz = <4000000>;
+	i3c-od-scl-hz = <1500000>;
```
Comment (MaureenHelm) [inline samples/sensor/thermometer/boards/frdm_mcxn947_mcxn947_cpu0.overlay:18]: consider using `DT_FREQ_K`

File: samples/sensor/thermometer/sample.yaml [hunk 5]
```diff
@@ -10,3 +10,5 @@ tests:
       - frdm_k22f             # tcn75a
       - robokit1              # ntc_thermistor
       - adi_eval_adin1110ebz  # adt7420
+      - frdm_mcxn947/mcxn947/cpu0 # p3t1755
+      - frdm_mcxn947/mcxn947/cpu0/qspi # p3t1755
```
Comment (MaureenHelm) [inline samples/sensor/thermometer/sample.yaml:14]: Pick one. This list of boards tries to cover different sensor drivers, not different boards.

File: drivers/sensor/nxp/p3t1755/p3t1755.c (hunk 2 above)
Comment (mmahadevan108) [reply drivers/sensor/nxp/p3t1755/p3t1755.c:23]: Thanks Maureen, I have made a change to use function pointers and assigning it on a per instance basis.

File: drivers/sensor/nxp/p3t1755/p3t1755.c (hunk 3 above)
Comment (mmahadevan108) [reply drivers/sensor/nxp/p3t1755/p3t1755.c:120]: Sure, I have made this change.

File: samples/sensor/thermometer/sample.yaml (hunk 5 above)
Comment (mmahadevan108) [reply samples/sensor/thermometer/sample.yaml:14]: I have made the change.

**Review by MaureenHelm — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Pull request merged**